www/js/lib/uiUtil.js

```diff
@@ -32,6 +32,40 @@ if (webpMachine) {
 }
 
 define(rqDef, function(settingsStore) {
+
+    /**
+     * Displays a Bootstrap alert or confirm dialog box depending on the options provided
+     * 
+     * @param {String} label The modal's label or title which appears in the header 
+     * @param {String} message The alert message to display in the body of the modal 
+     * @param {Boolean} isConfirm If true, the modal will be a confirm dialog box, otherwise it will be an alert 
+     * @param {String} declineButtonText The text to display on the decline button (optional, Default = "Cancel") 
+     * @param {String} approveButtonText  The text to display on the approve button (optional, Default = "Confirm")
+     * @returns {Promise<Boolean>} A promise which resolves to true if the user clicked Confirm, false if the user clicked Cancel, backdrop or the cross(x) button
+     */
+    function systemAlert(label, message, isConfirm, declineButtonText, approveButtonText) {
+        declineButtonText = declineButtonText || "Cancel";
+        approveButtonText = approveButtonText || "Confirm";
+        return new Promise(function (resolve, reject) {
+            if (!label || !message) reject("Missing parameters");
+            document.getElementById("declineModal").innerHTML = declineButtonText;
+            document.getElementById("approveModal").innerHTML = approveButtonText;
+            document.getElementById("modalLabel").innerHTML = label;
```

Comment:
Avoid using `.innerHTML`: it can raise security issues.
Using `.textContent` instead should work here (untested)